### PR TITLE
fix: remove network port from tor address and replace with wallet name in maker identifier 

### PR DESCRIPTION
**Pull request by Ferryx349**
# Pull Request

## Description
Removes the network port from Tor address, and replaces with wallet name in maker identifier.
## Related Issue(s)
fixes - #797 #798

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactor / performance improvement
- [ ] Documentation update only
- [ ] CI / Docker / Build changes
- [ ] Other (please describe):

## Protocol Version(s) Affected
- [ ] Legacy (ECDSA — `messages.rs`, `contract.rs`, `handlers.rs`)
- [ ] Taproot-Musig2 (`messages2.rs`, `contract2.rs`, `handlers2.rs`)
- [x] Both
- [ ] Neither (infrastructure / tooling only)

## Affected Component(s)
- [x] **makerd** (background daemon)
- [x] **taker** (CLI client)
- [x] **maker-cli** (command-line tool)
- [x] Core protocol / cryptography / library
- [ ] **watch_tower** (contract monitoring & recovery)
- [ ] Docker / deployment scripts
- [x] Tests / test framework
- [ ] Documentation (`docs/`)
- [ ] Other (please specify):

### Code Quality
- [x] I ran `cargo +nightly fmt --all` and committed the result
- [x] I ran `cargo +stable clippy --all-features --lib --bins --tests -- -D warnings` with zero warnings
- [x] I ran `cargo +stable clippy --examples -- -D warnings` with zero warnings
- [x] I ran `RUSTDOCFLAGS="-D warnings" cargo +nightly doc --all-features --document-private-items --no-deps` with zero warnings
- [x] Pre-commit git hook passes (`ln -s ../../git_hooks/pre-commit .git/hooks/pre-commit` if not already set)

### Testing
- [x] All unit tests pass (`cargo test`)
- [x] Integration tests pass (`cargo test --features integration-test`)
  - [x] `abort1` - Full swap protocol with fidelity bonds
  - [x] `fidelity` - Bond creation and management
  - [x] `standard_swap` - Complete coinswap flow
  - [x] `fidelity_renewal` - Auto-renewal with port-less addresses
- [x] Changes were manually tested on **regtest**
- [x] End-to-end maker ↔ taker swap tested (where applicable)
- [x] Test-only code is gated behind `#[cfg(feature = "integration-test")]`

### Documentation
- [ ] Relevant files in the `docs/` folder were updated
- [x] Complex logic is commented

### Security & Privacy (Critical)
- [x] This change preserves **trustlessness** and **atomic swap guarantees**
- [x] No regression in **sybil resistance** or **fidelity bond** logic
- [x] Tor anonymity and P2P message flow were reviewed
- [x] No new attack vectors or trust assumptions introduced
- [x] Edge cases and error handling considered
- [x] ZMQ subscription integrity (raw tx/block feed) is unaffected
- [x] `integration-test` feature flag is not reachable in production code paths

## How to Test

```bash
# Run core fidelity tests
cargo test --test fidelity --all-features -- --nocapture
cargo test --test fidelity_renewal --all-features -- --nocapture

# Standard swap (good baseline)
cargo test --test standard_swap --all-features -- --nocapture

# Abort scenarios (verify protocol safety)
cargo test --test abort1 --all-features -- --nocapture
cargo test --test abort2_case1 --all-features -- --nocapture
cargo test --test taproot_taker_abort1 --all-features -- --nocapture

# Malicious behavior & recovery
cargo test --test malice1 --all-features -- --nocapture
cargo test --test malice2 --all-features -- --nocapture

# Full integration suite
cargo test --all-features -- --test-threads=1

# Code quality checks
cargo +nightly fmt --check
cargo +stable clippy --all-features -- -D warnings




<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Changes to Address Handling**
  * Maker addresses now use port-less format for fidelity proof verification
  * Onion address parsing now accepts optional port numbers (defaults to 6102)
  * Enhanced address validation to support both port and port-less address formats
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: src/maker/server.rs [hunk 1]
```diff
@@ -52,7 +52,7 @@ fn network_bootstrap(maker: Arc<Maker>) -> Result<String, MakerError> {
         } else {
             // Always Tor otherwise
             let maker_hostname = maker.get_tor_hostname()?;
-            format!("{maker_hostname}:{maker_port}")
+            maker_hostname.to_string()
```
[the remark this answers is not in the document]
Comment (Ferryx349) [reply src/maker/server.rs]: Fidelity proofs use hostname only; Tor connections default to port 6102. This separates verification from network discovery.

File: src/maker/server2.rs [hunk 2]
```diff
@@ -43,13 +43,11 @@ fn network_bootstrap_taproot(maker: Arc<Maker>) -> Result<String, MakerError> {
...
         // Always clearnet in integration tests
-        let maker_address = format!("127.0.0.1:{maker_port}");
-        maker_address
+        format!("127.0.0.1:{maker_port}")
     } else {
         // Always Tor in production
         let maker_hostname = maker.get_tor_hostname()?;
-        let maker_address = format!("{maker_hostname}:{maker_port}");
-        maker_address
+        maker_hostname.to_string()
```
[the remark this answers is not in the document]
Comment (Ferryx349) [reply src/maker/server2.rs]: ditto

File: src/maker/server.rs [hunk 3]
```diff
@@ -50,9 +50,9 @@ fn network_bootstrap(maker: Arc<Maker>) -> Result<String, MakerError> {
             // Always clearnet in integration tests
             format!("127.0.0.1:{maker_port}")
         } else {
-            // Always Tor otherwise
+            // Always Tor otherwise - use port-less for OP_RETURN
             let maker_hostname = maker.get_tor_hostname()?;
-            format!("{maker_hostname}:{maker_port}")
+            maker_hostname.to_string()
```
[the remark this answers is not in the document]
Comment (Ferryx349) [reply src/maker/server.rs]: ditto

**Comment by Ferryx349** [comment]
Screenshot 2026-04-03 at 15 31 08

removed netwok port from maker and replaced it with wallet name, explored onion address, fidelity bond outpoint but found they might not be available from start which seems messy wrt our scope..

though it adds redundancy as log already contains wallet-name, but for now it seemed to be better option, happy to replace it with some hardcoded stuff if needed related to our protocol, pls review

**Comment by muchai254** [comment]
Hey @abhay349, I agree with you here that using onion address and fidelity bond outpoint as the maker identifier may not be the best approach as they are not available at startup. Using the wallet name appears to be a good approach as it is available before anything else happens and seems to have the smallest implementation footprint, as seen on your PR.

However, this approach introduces a problem. All the makers will end up showing the default name `maker`. I first thought that this would be eliminated when one changed their name in the config files but unfortunately this is still overwritten with the defaults as explained by the comment here `src/maker/api.rs:241–244`: 

``` 
// Runtime fields — not read from config file
            data_dir: default_config.data_dir,
            network: default_config.network,
            wallet_name: default_config.wallet_name,
```

I would suggest using a maker's `tweakable_point` as the identifier. It is a compressed secp256k1 public key tied to the maker. It is unique as no two makers can share it not unless they share the same wallet and is also available on startup. The first 8 hex characters would be sufficient for this purpose. 

@mojoX911, what do you think of this approach? I would like to give it a try especially now that the new release is out.

**Comment by Ferryx349** [comment]
@muchai254 thanks a lot for informing , in fact i have found that issue in last days, **tweakable_point** here is definately a good option as it is already in the protocol and requires zero extra work, i did consider it as well while exploring for identifiers, but i found out If we ever rotate swap keys, change derivation paths, or add a new protocol variant, the maker's identity anchor silently changes with it which is not something we want from an identifier.
so i have transformed my work towards **Wallet fingerprint** as it is a BIP-32 standard identifier specifically designed for wallet identification, already available in the codebase via **master_key.fingerprint()**, costs nothing to access. More importantly, it stays stable across protocol changes, it is anchored to the wallet, not to swap operations.

**Comment by Ferryx349** [comment]
The wallet fingerprint is already implemented in **get_core_wallet_label()** and actively used in the RPC layer for Bitcoin core address labeling.
```
let address_label = address_label.unwrap_or(self.get_core_wallet_label());
```
To use it for maker identification, the only change required is making **get_core_wallet_label()** publicly accessible, no protocol changes, no security risk.

@mojoX911 can i proceed?

**Comment by mojoX911** [comment]
Fixed in #836
